location[] location_list
float32 temperature